lib/broccoli/glimmer-app.ts

```diff
@@ -278,19 +283,40 @@ export default class GlimmerApp extends AbstractBuild {
     return tsconfig ? { tsconfig } : DEFAULT_TS_OPTIONS;
   }
 
-  /**
-   * Creates a Broccoli tree representing the compiled Glimmer application.
-   *
-   * @param options
-   */
-  public toTree() {
-    let jsTree = this.javascriptTree();
-    let cssTree = this.cssTree();
-    let publicTree = this.publicTree();
-    let htmlTree = this.htmlTree();
-    let missingPackages = [];
+  private javascript() {
+    let { src, nodeModules } = this.trees;
+    let tsConfig = this.tsOptions();
+    let configTree = this.buildConfigTree(src);
+    let srcWithoutHBSTree = new Funnel(src, {
+      exclude: ['**/*.hbs', '**/*.ts']
+    });
+
+    // Compile the TypeScript and Handlebars files into JavaScript
+    let compiledHandlebarsTree = this.compiledHandlebarsTree(src);
+    let combinedConfigAndCompiledHandlebarsTree = new MergeTrees([configTree, compiledHandlebarsTree]);
+
+    // the output tree from typescript only includes the output from .ts -> .js transpilation
+    // and no other files from the original source tree
+    let source = this.compiledTypeScriptTree(combinedConfigAndCompiledHandlebarsTree, nodeModules, tsConfig);
+
+    // Merge the JavaScript source and generated module map and resolver
+    // configuration files together, making sure to overwrite the stub
+    // module-map.js and resolver-configuration.js in the source tree with the
+    // generated ones.
+    source = new MergeTrees([srcWithoutHBSTree, source, configTree], { overwrite: true });
```

Comment:
We should do a followup issue to see if we can remove `configTree` from this merge...